Update README for uv 0.3.0 (#6165)
astral-sh/uv: README.md

  </picture>
</p>

<p align="center">
  <i>Installing the Trio dependencies with a warm cache.</i>

> </p>

Reviewer: Why remove?

Author: Not intentional, should be replaced with new benchmarks.

- ⚡️ [10-100x faster](https://github.com/astral-sh/uv/blob/main/BENCHMARKS.md) than `pip`.
- 🐍 [Installs and manages](#python-management) Python versions.
- 🛠️ [Runs and installs](#tool-management) Python applications.
- ❇️ [Runs scripts](https://docs.astral.sh/uv/guides/scripts), with support for
  [inline dependency metadata](https://docs.astral.sh/uv/guides/scripts#declaring-script-dependencies).
- 🗂️ Provides [comprehensive project management](#project-management), with a
  [universal lockfile](https://docs.astral.sh/uv/concepts/projects#lockfile).

> - 🔩 Includes a [pip-compatible interface](#the-pip-interface) for a performance boost with a
>   familiar CLI.

Reviewer 2: Maybe something like this?

```suggestion
- 🔩 Includes a [pip-compatible interface](#the-pip-interface) for a drop-in performance boost.
```

Author: Haha "drop-in" is dangerous. But yeah... maybe.

Reviewer 2: I think we say drop-in right now..

Author: Yeah but it's been problematic

- 🏢 Supports Cargo-style [workspaces](https://docs.astral.sh/uv/concepts/workspaces) for scalable
  projects.
- 💾 Disk-space efficient, with a [global cache](https://docs.astral.sh/uv/concepts/cache) for
  dependency deduplication.

> - ⏬ Installable without Rust or Python via `curl` or `pip`.

Reviewer 3: Is it just me or does this read a bit self-contradicting: *Installable without Python via `pip`.*

Author: Haha this did come to mind.. but we also want to be able to tell people it's available by `pip` because it's such a popular installation method.. hm.

  </picture>
</p>

## Highlights

- 🚀 A single tool to replace `pip`, `pip-tools`, `pipx`, `poetry`, `pyenv`, `virtualenv`, and more.
- ⚡️ [10-100x faster](https://github.com/astral-sh/uv/blob/main/BENCHMARKS.md) than `pip`.
- 🐍 [Installs and manages](#python-management) Python versions.
- 🛠️ [Runs and installs](#tool-management) Python applications.

> - ❇️ [Runs scripts](https://docs.astral.sh/uv/guides/scripts), with support for

Reviewer: Should we say "single-file scripts"?

Author: Sure, that copy makes sense elsewhere.. I think.
